runtime/vam/expr/function/parse.go

```diff
@@ -0,0 +1,48 @@
+package function
+
+import (
+	"github.com/brimdata/super"
+	samfunc "github.com/brimdata/super/runtime/sam/expr/function"
+	"github.com/brimdata/super/vector"
+	"github.com/brimdata/super/zcode"
+)
+
+// https://github.com/brimdata/super/blob/main/docs/language/functions.md#parse_uri
+type ParseURI struct {
+	zctx    *super.Context
+	typ     super.Type
+	builder zcode.Builder
+}
+
+func newParseURI(zctx *super.Context) *ParseURI {
+	return &ParseURI{zctx: zctx, typ: samfunc.URIType(zctx)}
+}
+
+func (p *ParseURI) Call(args ...vector.Any) vector.Any {
+	vec := vector.Under(args[0])
+	if vec.Type().ID() != super.IDString {
+		return vector.NewWrappedError(p.zctx, "parse_uri: string arg required", args[0])
+	}
+	errMsgs := vector.NewStringEmpty(0, nil)
+	var errs []uint32
+	vb := vector.NewBuilder(p.typ)
+	for i := range vec.Len() {
+		s, null := vector.StringValue(vec, i)
+		if null {
+			vb.Write(nil)
+		} else {
+			p.builder.Truncate()
+			if err := samfunc.EncodeURI(&p.builder, s); err != nil {
+				errMsgs.Append(err.Error())
+				errs = append(errs, i)
+				continue
+			}
+			vb.Write(p.builder.Bytes())
+		}
+	}
+	out := vb.Build()
+	if len(errs) > 0 {
+		return vector.Combine(out, errs, vector.NewVecWrappedError(p.zctx, errMsgs, vector.NewView(args[0], errs)))
+	}
+	return out
```

Comment:
Nit: If samfunc.EncodeURI always returned a super.Value, you could simplify this to
```go
	nullType := samfunc.URIType()
	b := NewDynamicBuilder()
	for i := range vec.Len()
		var val super.Value
		s, null := vector.StringValue(vec, i)
		if null {
			val = super.NewValue(nullType, nil)
		} else {
			val = samfunc.EncodeURI(s)
		}
		b.Write(val)
	}
	return b.Build()
```